core/word_eval.py

```diff
@@ -304,8 +300,8 @@ def _ApplyPrefixOp(self, val, op_id):
       if val.tag == value_e.Str:
         length = len(val.s)
       elif val.tag == value_e.StrArray:
-        # TODO: There can be empty placeholder values in the array.
-        length = len(val.strs)
+        # There can be empty placeholder values in the array.
+        length = sum(s is not None for s in val.strs)
```

Comment:
I agree that it would be more consistent, but Python disagrees: `TypeError: object of type 'generator' has no len()`. You'd have to use `len([s for s in val.strs if s is not None])` and creating a list only to get the length seems a bit wasteful. So I tend to use punning as a counting idiom. It could be written `sum(1 for s in val.strs if s is not None)`.

Reply:
Ah OK, thanks for the correction.  I like the `1` idiom!